_docs/ux-design.md

```diff
@@ -0,0 +1,138 @@
+This is a living document bound to change as the product is being built,
+matures, and receives user feedback.
+
+## Intended Purpose
+
+The OpenTelemetry Go compile-time instrumentation tool allows users to
+instrument their full application automatically, without having to perform any
+significant code change beyond adding simple, well-defined configuration that
+may be as simple as adding the relevant tool dependencies.
+
+The main value proposition is:
+- Very low effort required for holistic instrumentation
+- Ability to instrument within third-party dependencies
+- Keeping the codebase completely de-coupled from instrumentation
+
+## Target Audience
+
+Compile-time instrumentation, like other techniques of automatic
+instrumentation, does not affort users the same amount of control over their
+instrumentation as manual instrumentation; it trades very granual control for
+significantly reduced implementation effort. As a result, compile-time
+instrumentation may not appeal to developers who have very specific requirements
+on what their instrumentation produces.
+
+The primary audience for the OpenTelementry Go compile-time instrumentation tool
+is composed of the following personas:
+
+- Application developers looking for a no-frills, turney instrumentation
+  solution
+- System operators look to instrument applications without involving developers
+
+The tool may however also be relevant to the following personas:
+- Security personnel looking for maximal instrumentation coverage
+- Library developers looking to improve the instrumentation experience for their
+  library
+
+Large applications and deployments may involve multiple of these personas,
+possibly including some that are not part of the primary audience. This is
+particularly true for Enterprise scale companies, where different parts of the
+organization are often involved at different stages of the software delivery
+lifecycle. It is hence important the OpenTelemetry Go compile-time
+instrumentation tool allows co-operation between each of these entities without
+coupling them too tightly.
+
+## User Experience Overview
+
+The OpenTelemetry Go compile-time instrumentation tool is composed of the
+following software artifacts (in the form of Go packages):
+
+- `github.com/open-telemetry/opentelemetry-go-compile-instrumentation/cmd/gotel`,
+  a command-line tool that can be installed using `go install` or using
+  `go get -tool`;
+- `github.com/open-telemetry/opentelemetry-go-compile-instrumentation/runtime`,
+  a small, self-contained package that contains essential runtime functionality
+  used by instrumented applications (not intended for manual usage).
+
+### Getting Started
+
+The tool offers a wizard for getting started with the tool automatically by
+following a series of prompts:
+
+```console
+$ go run github.com/open-telemetry/opentelemetry-go-compile-instrumentation/cmd/gotel@latest setup
```

Comment:
That's a similar approach to what we have done for PHP some time ago in [this repository](https://github.com/open-telemetry/opentelemetry-php-contrib/tree/main/src/AutoInstrumentationInstaller). Nice to see a similar thought process.